---
layout: page
title: Index to topics covered in 2024 (EEB 5349)
permalink: /index2024/
---
Return to the [Phylogenetics (EEB 5349) main page](/phylogenetics2024/).

The following topics were covered in lecture in the Spring 2024 version of EEB 5349 (Phylogenetics). Please refer to the [schedule](/phylogenetics2024/) to find links to the slides for the dates indicated.

adaptive radiations, Apr-23

Akaike Information Criterion (AIC), Feb-29

among-site state frequency heterogeneity, Mar-21

among-site rate heterogeneity (ASRV), Feb-8

archery priors example, Feb-20

ascertainment bias, Mar-21

autoapomorphy trail, Mar-21

Bayesian analysis of macroevolutionary mixtures (BAMM), Apr-23

bat monophyly test, Feb-15

Bayes' rule, Feb-20

Bayesian Information Criterion (BIC), Feb-29

binary-state speciation and extinction (BiSSE) model, Apr-23

birth-death tree model, Apr-16

bootstrapping (non-parametric), Feb-15

bootstrapping (parametric), Mar-26

bootstrapping (IQ-TREE ultrafast), Feb-15

branch-and-bound, Jan-18 

Brownian motion model, Apr-2

calibration prior, Apr-16

coalescence, Mar-7

codon models, Apr-11

columbine example, Apr-23

comparative method, Apr-2

concordance factor (CF), Apr-9

conditioning on variability, Mar-21

confidence intervals, Apr-16

consensus trees, Jan-18 

correlated relaxed clock model, Apr-16

covarion, Apr-18

credible intervals, Apr-16

deep coalescence, Mar-7

delta, Pagel's, Apr-4

density example (gold-to-aluminum brick), Feb-20

dependence model, Mar-26

Dirichlet process prior, Apr-9

discrete Gamma model, Feb-8

distribution, Beta, Mar-5, Mar-21

distribution, Dirichlet, Mar-5

distribution, discrete Uniform, Mar-5

distribution, Gamma, Feb-8, Mar-5

distribution, Gamma-Dirichlet, Mar-5

distribution, Lognormal, Mar-5

distribution, marginal posterior, Feb-27

distribution, proposal (for MCMC), Feb-22

distribution, support, Mar-5

distribution, target (for MCMC), Feb-22

diversification rate, Apr-23

dwell time, Mar-28

edge length, Jan-30

effective population size (Ne), Mar-19

estrus advertisement example, Mar-26

evolutionary distance, Feb-6

exchangeabilities, nucleotide, Jan-30

exchangeabilities, amino acid, Apr-11

exhaustive search, Jan-18 

expected time to coalescence, Mar-7

extinction rate, Apr-23

evolutionary correlation, Mar-26

Felsenstein (1981) model (F81), Feb-1

Felsenstein (1984) model (F84), Feb-1

fence analogy (truncated priors), Mar-5

fossilized birth-death process, Apr-16

free rates model (+R), Feb-15

frequencies, equilibrium nucleotide, Jan-30

frequencies, equilibrium amino acid, Apr-11

frequencies, equilibrium state, Mar-21

gamma, Pybus and Harvey, Apr-23

gene trees, Mar-7

general time-reversible model (GTR), Feb-1

genetic code, Apr-11

Hasegawa, Kishino, and Yano (1985) model (HKY85), Feb-1

heated chains (MCMC), Feb-22

heterotachy, Feb-8, Apr-18

hyperparameter, Mar-5

hyperprior, Mar-5

independence of events, Feb-6

induced prior, Mar-5

integrals, Feb-20

invariable sites model (+I), Feb-8

Jukes-Cantor (1969) model (JC69), Jan-30

K, Blomberg's, Apr-4

kernel density estimation, Feb-27

Kimura (1980) 2-parameter model (K2P, K80), Feb-1

lambda, Pagel's, Apr-4

likelihood criterion, Jan-18 

likelihood (single sequence), Feb-6

likelihood (evolutionary distance), Feb-6

likelihood ratio test, Feb-6

likelihood, vs. probability, Feb-20

lineage-through-time (LTT) plots, Apr-23

marginal likelihood, Feb-20, Feb-29

marginal likelihood (steppingstone method, Feb-29

Markov model, Jan-30

maximum agreement subtree, Jan-18
 
maximum likelihood estimate (MLE), Feb-6

Markov-chain Monte Carlo (MCMC), Feb-22

MCMC, robot metaphor, Feb-22

MCMC, local move proposal, Feb-27

MCMC, reversible jump (rjMCMC used in BAMM, heterotachy), Mar-26, Apr-18, Apr-23

MCMC, sliding window proposal, Feb-27

Metropolis algorithm, Feb-22

Metropolis-Hastings algorithm, Feb-22

mixing (MCMC), Feb-22

goodness-of-fit of a model, Jan-30

Hastings ratio, Feb-22

models, amino acid, Apr-11

models, asymmetric, Mar-21

models, Brownian motion, Apr-2

models, codon, Apr-9

models, covarion, Apr-18

models, dependence, Mar-26

models, Dirichlet process prior, Apr-9, Apr-18

models, discrete Gamma, Feb-8

models, discrete morphology, Mar-21

models, empirical Bayesian, Mar-5

models, hierarchical, Mar-5

models, independence, Mar-26

models, mixture (+I, +G, Q matrix, edge length), Feb-8, Apr-18

models, mixture (C20), Apr-11

models, Mk, Mar-21

models, partition, Feb-8

models, protein, Apr-11

models, relaxed clock Apr-11, Apr-16

models, state-dependent diversification (BiSSE, MuSSE, GeoSSE, QuaSSE, HiSSE), Apr-23

models, symmetric, Mar-21

monophyly, Jan-16 

nearest-neighbor interchange (NNI) branch swapping, Jan-18 

newick trees, Jan-16 

nonsynonymous/synonymous rate ratio (omega), Apr-11

point accepted mutation (PAM) unit, Apr-11

paraphyly, Jan-18 

parsimony criterion, Jan-18 

partitioned models, Feb-8

paternity example, Feb-20

phylogenetically generalized least squares (PGLS), Apr-2

phylogenetically independent contrasts (PIC), Apr-2

polyphyly, Jan-18 

predictive ability of a model, Jan-30

prior, calibration, Apr-16

prior, coalescent tree, Mar-19

prior, distribution, Mar-5

prior, Gamma-Dirichlet, Mar-5

prior, induced split, Mar-5

prior, polytomy, Mar-26

probability, "and" rule, Feb-6

probability, conditional, Feb-6

probability, density, Feb-20

probability, joint, Feb-6

probability, marginal, Feb-6

probability, mass vs. density, Feb-20

probability, no coalescence, Mar-7

probability, "or" rule, Feb-6

probability, prior, Feb-20

probability, posterior, Feb-20

probability, support, Mar-5

probability, total, Feb-6

probability, vs. likelihood, Feb-20

pruning algorithm, Feb-8

pull of the present, Apr-23

random local clocks model, Apr-16

running on empty (MCMC without data), Mar-5

site-specific rates, Feb-8

software, ape, Apr-11

software, BayesTraits, Apr-4

software, BUCKy, Apr-9

software, caper, Apr-11

software, geiger, Apr-11

software, MrBayes, Mar-7

software, PAUP*, Jan-18 (and many times after)

software, PhyloBayes, Apr-9

software, phytools, Apr-11

software, RevBayes, Mar-21, Mar-28

speciation rate, Apr-23

splits, Jan-16 

state-dependent diversification models, Apr-23

steppingstone method (marginal likelihood estimation), Feb-29

stochastic character mapping, Mar-28

subway map model analogy, Jan-30

rare disease example, Feb-20

rate heterogeneity, Feb-8

rate matrix (instantaneous), Feb-1

reconnection limit, Jan-18 

simulation, Feb-15

sojourn time, Mar-28

star decomposition, Jan-18 

stepwise addition, Jan-18 

substitution rate, Jan-30

subtree-pruning-regrafting (SPR) branch swapping, Jan-18 

theta (population genetics parameter), Mar-7

tip dating, Apr-16

trace plots, Feb-22

traffic accidents Poisson example, Mar-21

transition probabilities, Jan-30

transition (substitution type), Feb-1

transition-transversion rate ratio, Feb-1

transition-transversion ratio, Feb-1

transversion (substitution type), Feb-1

tree islands, Jan-18 

tree-bisection-reconnection (TBR) branch swapping, Jan-18 

turnover, Apr-23

ultrametric, Jan-18 

uncorrelated lognormal relaxed clock model, Apr-16

variance-covariance matrix, Apr-2

Yule pure-birth tree model, Apr-16
