---
layout: page
title: Testing Page
permalink: /test/
---

## New _schedule_ design:

<table>
<tr><th>Date</th> <th>Lecture topic</th> <th>Lab/homework</th></tr>
<tr><td>Tuesday Jan. 16   </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/01-intro-annotated.pdf">Introduction</a>                                                </td><td> Homework 1: <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/hw1-TreeFromSplits.pdf">Splits</a> (due 2024-01-23)                                                            </td></tr>
<tr><td>Thursday Jan. 18  </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/02-searching-annotated.pdf">Optimality criteria, 
                                                                                                        search strategies, consensus trees</a>                                          </td><td> Lab 1: <a href="https://plewis.github.io/xanadu/">Using the Xanadu cluster</a>, <a href="/paup/">Introduction to PAUP*</a>, and <a href="/nexus/">NEXUS file format</a>                </td></tr>
<tr><td>Tuesday Jan. 23   </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/03-parsimony-annotated.pdf">The parsimony criterion</a>                                 </td><td> Homework: 2 <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/hw2-parsimony.pdf">Parsimony</a> (due 2024-01-30)                                                              </td></tr>
<tr><td>Thursday Jan 25   </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/04-distances-annotated.pdf">Distance methods</a>                                        </td><td> Lab 2: <a href="https://plewis.github.io/searching/">Searching</a>                                                                                                                     </td></tr>
<tr><td>Tuesday Jan. 30   </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/05-models-annotated.pdf">Substitution models</a>                                        </td><td> Homework: 3 <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/hw3-distance.pdf">Distances</a>, <a href="https://plewis.github.io/python/">Python primer</a> (due 2024-02-06) </td></tr>
<tr><td>Thursday Feb. 1   </td><td> Substitution models (cont.)                                                                                                                         </td><td> Lab 3: <a href="https://plewis.github.io/likelihood/">Estimating parameters using likelihood</a>                                                                                       </td></tr>
<tr><td>Tuesday Feb. 6    </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/06-likelihood-annotated.pdf">Maximum likelihood criterion</a>                           </td><td> Homework: 4 <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/04-hw4-k80beta.pdf">Transitions and transversions</a> (due 2024-02-15)                                         </td></tr>
<tr><td>Thursday Feb. 8   </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/07-ratehet-annotated.pdf">Rate heterogeneity </a>                                       </td><td> Lab 4: <a href="https://plewis.github.io/simulation/">Simulating sequence data</a>                                                                                                     </td></tr>
<tr><td>Tuesday Feb. 13	  </td><td> SNOW DAY (no class)                                                                                                                                 </td><td> Homework: 5 <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/hw5-likelihood.pdf">Site likelihood</a> (due 2024-02-20)                                                       </td></tr>
<tr><td>Thursday Feb. 15  </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/07-ratehet2-annotated.pdf">Rate het. (finish up)</a>, 
                                    <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/08-bootstrapping-annotated.pdf">Bootstrapping</a>, 
                                    <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/bootstrap.Rmd">Bootstrapping example in R</a>, 
                                    <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/08-simulation-annotated.pdf">Simulation</a>                                             </td><td> Lab 5: <a href="https://plewis.github.io/iqtree/">Maximum likelihood analyses with IQ-TREE</a>                                                                                         </td></tr>
<tr><td>Tuesday Feb. 20   </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/09-bayesrule-annotated.pdf">Bayes' Rule and Bayesian statistics</a>                     </td><td> Homework 6: <a href="https://plewis.github.io/hwratehet/">Rate Heterogeneity</a> (due 2024-02-27)                                                                                                        </td></tr>
<tr><td>Thursday Feb. 22  </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/10-mcmc-annotated.pdf">Markov chain Monte Carlo (MCMC)</a>                              </td><td> Lab 6: <a href="https://plewis.github.io/rprobdist/">Using R to explore probability distributions</a> (through Exponential, questions 1-19)                                            </td></tr>
<tr><td>Tuesday Feb. 27   </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/11-mcmc2-annotated.pdf">MCMC (cont.)</a>                                                </td><td> Homework 7: <a href="https://plewis.github.io/hwmcmc/">MCMC</a> (due 2024-03-05)                                                                                                       </td></tr>
<tr><td>Thursday Feb. 29  </td><td> <a href="https://gnetum.eeb.uconn.edu/courses/phylogenetics/12-model-selection.pdf">Model selection</a>                                             </td><td> Lab 7: <a href="https://plewis.github.io/rprobdist/">Using R to explore probability distributions</a> (start with Lognormal, questions 20-29)                                          </td></tr>
<tr><td>Tuesday Mar. 5    </td><td> Coalescence and gene trees vs. species trees                                                                                                        </td><td> Homework 8</td></tr>
<tr><td>Thursday Mar. 7   </td><td> Species tree estimation                                                                                                                             </td><td> Lab 8:</td></tr>
<tr><td>Tuesday Mar. 12   </td><td> SPRING BREAK                                                                                                                                        </td><td></td></tr>
<tr><td>Thursday Mar. 14  </td><td> SPRING BREAK                                                                                                                                        </td><td></td></tr>
<tr><td>Tuesday Mar.  19  </td><td> Codon, secondary structure, and amino acid models                                                                                                   </td><td> Homework 9</td></tr>
<tr><td>Thursday Mar. 21  </td><td> Discrete morphological models                                                                                                                       </td><td> Lab 9</td></tr>
<tr><td>Tuesday Mar. 26   </td><td> Polytomies; Pagel's test                                                                                                                            </td><td> Homework 10</td></tr>
<tr><td>Thursday Mar. 28  </td><td> Stochastic character mapping                                                                                                                        </td><td> Lab 10</td></tr>
<tr><td>Tuesday Apr. 2    </td><td> Evolutionary Correlation: Continuous Traits                                                                                                         </td><td> Homework 11</td></tr>
<tr><td>Thursday Apr. 4   </td><td> PGLS regression                                                                                                                                     </td><td> Lab 11</td></tr>
<tr><td>Tuesday Apr. 9    </td><td> Phylogenetic signal in continuous traits                                                                                                            </td><td> Homework 12</td></tr>
<tr><td>Thursday Apr. 11  </td><td> Multispecies coalescent and species tree estimation                                                                                                 </td><td> Lab 12</td></tr>
<tr><td>Tuesday Apr. 16   </td><td> Network models                                                                                                                                      </td><td> Homework 13</td></tr>
<tr><td>Thursday Apr. 18  </td><td> Diversification rate evolution                                                                                                                      </td><td> Lab 13</td></tr>
<tr><td>Tuesday Apr. 23   </td><td> Diversification (cont.), Heterotachy, and Covarion models                                                                                           </td><td> no homework assignment</td></tr>
<tr><td>Thursday Apr. 25  </td><td> Species delimitation and information                                                                                                                </td><td> Lab 14</td></tr>
</table>